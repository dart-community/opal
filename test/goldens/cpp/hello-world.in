#include <iostream>
#include <string_view>

constexpr std::string_view message = "Hello, world!";

int main() {
  // Print the message.
  std::cout << "The message is: " << message << std::endl;

  return 0;
}
