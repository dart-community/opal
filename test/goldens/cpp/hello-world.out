#include <iostream>
^
source-code-cpp-metadata-preprocessor-directive-punctuation
 ^^^^^^^
 source-code-cpp-metadata-preprocessor-directive-metadata-preprocessor-directive-name
        ^^^^^^^^^^^
        source-code-cpp-metadata-preprocessor-directive-metadata-preprocessor-directive-options
#include <string_view>
^
source-code-cpp-metadata-preprocessor-directive-punctuation
 ^^^^^^^
 source-code-cpp-metadata-preprocessor-directive-metadata-preprocessor-directive-name
        ^^^^^^^^^^^^^^
        source-code-cpp-metadata-preprocessor-directive-metadata-preprocessor-directive-options

constexpr std::string_view message = "Hello, world!";
^^^^^^^^^
source-code-cpp-keyword-modifier-constexpr
         ^
         source-code-cpp-whitespace
          ^^^
          source-code-cpp-identifier
             ^^
             source-code-cpp-keyword-operator
               ^^^^^^^^^^^
               source-code-cpp-identifier-type-built-in
                          ^
                          source-code-cpp-whitespace
                           ^^^^^^^
                           source-code-cpp-identifier
                                  ^
                                  source-code-cpp-whitespace
                                   ^
                                   source-code-cpp-keyword-operator
                                    ^
                                    source-code-cpp-whitespace
                                     ^
                                     source-code-cpp-literal-string-quoted-double-literal-string-quoted-double-begin
                                      ^^^^^^^^^^^^^
                                      source-code-cpp-literal-string-quoted-double-literal-string-unquoted-literal-string-unquoted
                                                   ^
                                                   source-code-cpp-literal-string-quoted-double-literal-string-quoted-double-end
                                                    ^
                                                    source-code-cpp-punctuation

int main() {
^^^
source-code-cpp-identifier-type-built-in
   ^
   source-code-cpp-whitespace
    ^^^^
    source-code-cpp-identifier-function
        ^
        source-code-cpp-punctuation
         ^
         source-code-cpp-punctuation
          ^
          source-code-cpp-whitespace
           ^
           source-code-cpp-punctuation
  // Print the message.
^^
source-code-cpp-whitespace
  ^^^^^^^^^^^^^^^^^^^^^
  source-code-cpp-comment-line
  std::cout << "The message is: " << message << std::endl;
^^
source-code-cpp-whitespace
  ^^^
  source-code-cpp-identifier
     ^^
     source-code-cpp-keyword-operator
       ^^^^
       source-code-cpp-identifier
           ^
           source-code-cpp-whitespace
            ^^
            source-code-cpp-keyword-operator
              ^
              source-code-cpp-whitespace
               ^
               source-code-cpp-literal-string-quoted-double-literal-string-quoted-double-begin
                ^^^^^^^^^^^^^^^^
                source-code-cpp-literal-string-quoted-double-literal-string-unquoted-literal-string-unquoted
                                ^
                                source-code-cpp-literal-string-quoted-double-literal-string-quoted-double-end
                                 ^
                                 source-code-cpp-whitespace
                                  ^^
                                  source-code-cpp-keyword-operator
                                    ^
                                    source-code-cpp-whitespace
                                     ^^^^^^^
                                     source-code-cpp-identifier
                                            ^
                                            source-code-cpp-whitespace
                                             ^^
                                             source-code-cpp-keyword-operator
                                               ^
                                               source-code-cpp-whitespace
                                                ^^^
                                                source-code-cpp-identifier
                                                   ^^
                                                   source-code-cpp-keyword-operator
                                                     ^^^^
                                                     source-code-cpp-identifier
                                                         ^
                                                         source-code-cpp-punctuation

  return 0;
^^
source-code-cpp-whitespace
  ^^^^^^
  source-code-cpp-keyword-control-return
        ^
        source-code-cpp-whitespace
         ^
         source-code-cpp-literal-number-integer
          ^
          source-code-cpp-punctuation
}
^
source-code-cpp-punctuation
